primitives/sr-primitives/src/lib.rs

```diff
@@ -348,56 +348,41 @@ impl From<ed25519::Signature> for AnySignature {
 
 #[derive(Eq, PartialEq, Clone, Copy, Decode, Encode, RuntimeDebug)]
 #[cfg_attr(feature = "std", derive(Serialize))]
-/// Reason why an extrinsic couldn't be applied (i.e. invalid extrinsic).
-pub enum ApplyError {
-	/// General error to do with the permissions of the sender.
-	NoPermission,
-
-	/// General error to do with the state of the system in general.
-	BadState,
-
+/// Reason why an extrinsic couldn't be included into a block.
+pub enum InclusionError {
 	/// Any error to do with the transaction validity.
 	Validity(transaction_validity::TransactionValidityError),
```

Comment:
If the enum just has one variant, we should drop it and just use `TransactionValidityError`.